### DISCUSSION TITLE
some fix always failed

### DISCUSSION THREAD

**Discussion by Tean** [usnistgov/macos_security#233] [Q&A] [answered]
Like system_settings_bluetooth,.
I try run the compliance script with '--check' and '--fix', then I try run the script
`result_value=$(/usr/bin/osascript -l JavaScript << EOS
$.NSUserDefaults.alloc.initWithSuiteName('com.apple.MCXBluetooth')\
.objectForKey('DisableBluetooth').js
EOS
)`
the result_value is ''
After I looked for the fix script part, but I can't find it.
So Its means I need disable bluetooth manually(not turn off, but I can't find way to disable it)?  Or what am I missing?
ps: I run the 'all_rules' check, after --fix, have about 100+ failed 

My device:  Macbook Pro 2020 13-inch M1, OS version is Ventura 13.2.1, and I checkout the correct branch(ventura)

**Comment by golbiga** [comment, marked as answer]
Those failures are due to the fact that you're not applying the configuration profiles to that device. That specific check you posted above is checking the `com.apple.MCXBluetooth` payload for the `DisableBluetooth` key. When you run the generate guidance script add `-p` and it will create all the profiles for you. One thing to note, I wouldn't use all_rules on a production system, there are some rules that can lead to locking yourself out of your system (smartcard controls).

**Reply by Tean** [reply]
Thanks, I got it.
